### ISSUE TITLE
Audio plugin missing interpolation

### ISSUE THREAD

**Issue by ghost** [Mystro256/wiisxr#2]
The current audio plugin on the recent build is, how do I say, no different than the previous WiiSX branch's, which is understandable. However, upon testing, I noted several issues with it: 
- Sound sync; Many sound effects are incorrect compared to default audio for PCSX-R
- ADSR/Reverb: Many games sound flat, notably Final Fantasy games
- Gaussian interpolation; I don't know how well the Wii can handle this being implemented, but this would help the audio sound less"raspy" if you will. 

Posted below are a few samples taken from PCSX-R as a test to show how the sound is:

Final Fantasy VII cursor (real hardware and PCSX sound the same):
http://filetrip.net/view?0NOICdUuWZ

Crash Bandicoot 3 sample: Listen to the cymbals and how they sound somewhat muffled, as per Gaussian interpolation does things (among the other sounds):
http://filetrip.net/view?bemsWwjC5e

Once more, these audio issues existed in previous forks of WiiSX and I hope I can do my part to help these issues better understood. Thank you for your time. 

Build I used https://github.com/Mystro256/wiisxr/raw/master/Gamecube/WiiSX.dol

**Comment by Mystro256** [comment]
Indeed, I plan to port the PCSXR sound plugin to this, if feasible. This should have the audio more in line with PCSXR.

To clarify, these were issues with the original WiiSX right? I have yet to do any work on the audio, so this should be the same. If this was an issue that did not exist on the official WiiSX 2.1 Beta build, please let me know, so I can fix this regression.

Thanks!

**Comment by roflcopter777** [comment]
As far as I recall, the issue has existed since the beginning, even up to version 2.1, the lack of ADSR ( I don't even know if they call it that with the PSX audio processor), but I do know that Gaussian is definitely missing along with sound sync, which is the cause for the off sound effects in Square Enix games. I'll be happy to test some more on this tomorrow as it is getting late here. That being said, I appreciate the work you're putting in to this, and would be happy to do some comparative ABX tests. 

Also, my other account was closed, so I'm the one who reported this issue :)

**Comment by Mystro256** [comment]
That's ok, I just wanted to make sure there wasn't any regression from the original code base. I'll see what I can do to make this happen.

Thanks

**Comment by roflcopter777** [comment]
That ain't good

**Comment by roflcopter777** [comment]
Yeaaah this isn't going be fixed I take  it

**Comment by Mystro256** [comment]
@roflcopter777 In case you are confused, I did not close this issue, just #12 , which was referenced, as it was stopping me from working on this issue (i.e. "blocking it").

I plan to finish this and XA, but keep in mind that I do not have a lot of time available, so it's going to take some time. If you or anyone else would like to take up programming and help me fix something, by all means.

Note that I've been working on more critical issues because I've had some extra time on my hands in the last week (my last change prior to last Sunday was back in April, if that puts things in perspective).

Graphics and audio improvements will take me time, so don't expect something overnight. Without help, it may take me forever, but I'm going to try to do it anyway.

**Comment by roflcopter777** [comment]
Apologies, I wasn't sure what was planned for 2.3,sorry -_-  Have you considered asking the ePSXe team for help with the audio?

**Comment by Mystro256** [comment]
It's ok, I plan to release 2.3 after I fix a few more things (likely #7 and #31).
Audio improvements will likely be targeted for a later release: hopefully sooner than later though. Note that I don't have any strict version schedule, it's just sort of an "as needed" situation and whenever I've fixed enough to justify making another release.

As for ePSXe, this project is licensed with GPLv2+, which can only be legally be mixed with other open source content. As a general rule, most of the ePSXe codebase is not GPL compatible. Unless there is an open source audio plugin that happens to work with ePSXe, it won't really be any help to wiisxr.

Furthermore, PCSXR is based on the same code, so it would be easier to use that as a starting point instead, especially because it's fully GPL compatible, which will avoid any copyright issues.

**Comment by roflcopter777** [comment]
Will there be improvements to the GPU plugin (vsync, to reduce screen tearing and maybe culling, etc?) as well? Not trying to rush ya, just curious :D   Looking forward to it

**Comment by Mystro256** [comment]
Indeed, see issue #4 
It will take a while, but it'll make it look nice.
